tests/src/Interop/AssemblyLoadContext30Extensions/AssemblyLoadContext30Extensions.cs

```diff
@@ -0,0 +1,234 @@
+    {
+        foreach (Assembly b in alc.Assemblies)
+        {
+            if (a == b)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static bool ContextsContainContext(AssemblyLoadContext alc)
+    {
+        foreach (AssemblyLoadContext c in AssemblyLoadContext.Contexts)
+        {
+            if (alc == c)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void DefaultName()
+    {
+        try
+        {
+            Console.WriteLine("DefaultName()");
+
+            AssemblyLoadContext alc = AssemblyLoadContext.Default;
+            Assert(alc == AssemblyLoadContext.GetLoadContext(typeof(Program).Assembly));
+            Assert(ContextsContainContext(alc));
+            Assert(AssembliesContainAssembly(alc, typeof(Program).Assembly));
+
+            Console.WriteLine(alc.Name);
+            Assert(alc.Name == "Default");
+
+            Console.WriteLine(alc.GetType().ToString());
+            Assert(alc.GetType().ToString() == "System.Runtime.Loader.DefaultAssemblyLoadContext");
+
+            Console.WriteLine(alc.ToString());
+            Assert(alc.ToString().Contains("\"Default"));
+            Assert(alc.ToString().Contains("\" System.Runtime.Loader.DefaultAssemblyLoadContext"));
+            Assert(alc.ToString().Contains(" #"));
+        }
+        catch (Exception e)
+        {
+            Assert(false, e.ToString());
+        }
+    }
+
+    public static void AssemblyLoadFileName()
+    {
+        try
+        {
+            Console.WriteLine("AssemblyLoadFileName()");
+
+            String path = typeof(Program).Assembly.Location;
+            Assembly a = Assembly.LoadFile(path);
+
+            Assert(a != typeof(Program).Assembly);
+
+            AssemblyLoadContext alc = AssemblyLoadContext.GetLoadContext(a);
+            Assert(ContextsContainContext(alc));
+            Assert(AssembliesContainAssembly(alc, a));
+            Assert(alc != AssemblyLoadContext.Default);
+
+
+            Console.WriteLine(alc.Name);
+            Assert(alc.Name == String.Format("Assembly.LoadFile({0})", path));
+
+            Console.WriteLine(alc.GetType().ToString());
+            Assert(alc.GetType().ToString() == "System.Runtime.Loader.IndividualAssemblyLoadContext");
+
+            Console.WriteLine(alc.ToString());
+            Assert(alc.ToString().Contains("\"" + String.Format("Assembly.LoadFile({0})", path)));
+            Assert(alc.ToString().Contains("\" System.Runtime.Loader.IndividualAssemblyLoadContext"));
+            Assert(alc.ToString().Contains(" #"));
+        }
+        catch (Exception e)
+        {
+            Assert(false, e.ToString());
+        }
+    }
+
+    public static void AssemblyLoadByteArrayName()
+    {
+#if ReadAllBytes // System.IO.File.ReadAllBytes is not found when ReferenceSystemPrivateCoreLib is true
```

Comment:
I'll fix this when the façade is added in corefx and we remove `<ReferenceSystemPrivateCoreLib>true</ReferenceSystemPrivateCoreLib>`